docs/rules/no-get-by-for-asserting-element-not-present.md

```diff
@@ -1,75 +1,58 @@
+However, when trying to assert if an element is not present or disappearance, using `getBy*` will make the test fail immediately. Instead it is recommended to use `queryBy*`, which does not throw and therefore we can assert that e.g. `expect(queryByText("Foo")).not.toBeInTheDocument()`.
 
 > The same applies for the `getAll*` and `queryAll*` queries too.
 
 ## Rule details
 
-This rule gives a notification whenever `expect` is used with one of the query functions that throw an error if the element is not found.
+This rule gives a notification whenever `expect` is used with one of the query functions that throws an error if the element is not found.
 
 Examples of **incorrect** code for this rule:
 
 ```js
 test('some test', () => {
   const { getByText, getAllByText } = render(<App />);
-  expect(getByText('Foo')).toBeInTheDocument();
-  expect(getAllByText('Foo')[0]).toBeInTheDocument();
   expect(getByText('Foo')).not.toBeInTheDocument();
-  expect(getAllByText('Foo')[0]).not.toBeInTheDocument();
+  expect(getAllByText('Foo')[0]).toBeNull();
+  expect(getByText('Foo')).toBeFalsy();
+  expect(getAllByText('Foo')[0]).not.toBeTruthy();
+  expect(getByText('Foo')).toBeUndefined();
 });
 ```
 
 ```js
 test('some test', async () => {
   const utils = render(<App />);
-  await wait(() => {
+  await waitForElementToBeRemoved(() => {
     expect(utils.getByText('Foo')).toBeInTheDocument();
   });
-  await wait(() => {
+  await waitForElementToBeRemoved(() => {
     expect(utils.getAllByText('Foo')).toBeInTheDocument();
   });
-  await wait(() => {
-    expect(utils.getByText('Foo')).not.toBeInTheDocument();
-  });
-  await wait(() => {
-    expect(utils.getAllByText('Foo')).not.toBeInTheDocument();
-  });
 });
 ```
 
 Examples of **correct** code for this rule:
 
 ```js
 test('some test', () => {
-  const { queryByText, queryAllByText } = render(<App />);
-  expect(queryByText('Foo')).toBeInTheDocument();
-  expect(queryAllByText('Foo')[0]).toBeInTheDocument();
-  expect(queryByText('Foo')).not.toBeInTheDocument();
-  expect(queryAllByText('Foo')[0]).not.toBeInTheDocument();
+  const { getByText, getAllByText } = render(<App />);
+  expect(getByText('Foo')).toBeInTheDocument();
+  expect(getAllByText('Foo')).toBeTruthy();
 });
 ```
 
 ```js
 test('some test', async () => {
   const utils = render(<App />);
-  await wait(() => {
-    expect(utils.queryByText('Foo')).toBeInTheDocument();
-  });
-  await wait(() => {
-    expect(utils.queryAllByText('Foo')).toBeInTheDocument();
-  });
-  await wait(() => {
-    expect(utils.queryByText('Foo')).not.toBeInTheDocument();
-  });
-  await wait(() => {
-    expect(utils.queryAllByText('Foo')).not.toBeInTheDocument();
-  });
+  await waitForElementToBeRemoved(() => utils.queryByText('Foo'));
+  await waitForElementToBeRemoved(() => utils.queryAllByText('Foo'));
```

Comment:
This one is wrong as you can't use a query returning arrays within `waitForElementToBeRemoved`, so this is another reason to remove `getAllBy` from this rule.